tests/compiler/test_type_hinting.py

```diff
@@ -83,29 +83,6 @@ def test_basic_type_hint_name_collision(snippetcompiler):
     (_, scopes) = compiler.do_compile()
 
 
-def test_advanced_type_hint_name_collision(snippetcompiler):
-    snippetcompiler.setup_for_error(
-        """
-import elaboratev1module
-
-entity A extends elaboratev1module::A:
-
-end
-
-A.ref [1] -- elaboratev1module::A
-
-A(ref=A())
-
-implement elaboratev1module::A using std::none
-implement A using std::none
-
-        """,
-        "Could not determine namespace for type A. 2 possible candidates exists: [__config__::A, elaboratev1module::A]. "
-        "To resolve this, use the fully qualified name instead of the short name. "
-        "(reported in Construct(A) ({dir}/main.cf:10:7))",
-    )
-
-
```

Comment:
I figure this test that is now failing can go now that we fixed the bug ?

Reply:
I think the test as it is now is wrong, but I think we do still need one that tests for this error behavior in a correct way. You can drop it for now (this bug is causing issues for solutions so the faster we merge it the better) but after we've reached a conclusion on Slack we should add a replacement in a second PR.

Reply:
I agree